### PR TITLE
Dictionary with page info now passed to template

### PR DESCRIPTION
**Pull request by astro-josh**
html_index now passes a list of lists of dictionaries with the keys: output_file_path, name, and title to the template for use. Each list is grouped by the subdir. Accepting these changes would require updating template file in notebooks repo to use new values passed.

### CODE REVIEW

**Comment by laurenmarietta** [comment]
A couple thoughts here:
1) It took me a while to get this to work at first, because `run_parsed` (and `process_notebooks`) only creates a **list of dictionaries** if it is re-processing notebooks. If the notebooks have already been executed, then `NBPagesConverter.convert` returns a **list of path strings** (`converter.py` line 136), which throws an error in `make_html_index` - see below for the error. I was ultimately able to get it to work by running `convert.py --overwrite`, but maybe something should be altered in `make_html_index`  or `NBPagesConverter.convert` so that that error doesn't come up if any notebooks have already been converted?
2) This might be unnecessary, but it would be easy to make this more backwards-compatible with the current `index.tpl` templates, just by keeping the `notebook_html_paths` keyword in the index renderer. So the `content` declaration would become: 
```python
content = templ.render(notebook_html_paths=converted_files, page_groups=converted_file_dicts)
```
Might be helpful so any repos that use the current version of `index.tpl` don't immediately break  when this PR is merged.

---
Error from `make_html_index`:
```
Traceback (most recent call last):
  File "convert.py", line 32, in <module>
    make_html_index(converted, './index.tpl')
  File "/Users/lchambers/CSI/nbpages/nbpages/html_index.py", line 43, in make_html_index
    dict(output_file_path=os.path.relpath(os.path.realpath(page['output_file_path']), outdir),
TypeError: string indices must be integers
```

**Comment by astro-josh** [comment]
@laurenmarietta Thanks! I've addressed the issues you've found.

@eteq I believe this is ready now.

File: nbpages/converter.py [hunk 1]
```diff
@@ -179,7 +179,18 @@ def convert(self, remove_executed=False):
...
+        try:
+            with open(self.nb_path) as f:
+                nb = nbformat.reader.read(f)
+                title = nb['cells'][0]['source'].split('#')[1].split("\n")[0].strip()
+        except Exception:
+            print('Failed to parse notebook title from first cell, please check notebook.')
+
+        page_info = dict(output_file_path=output_file_path, name=self.nb_name.replace("_", ' ').title(), title=title)
+
+        return (page_info)
```
Comment (eteq) [inline nbpages/converter.py]:
```suggestion
        return page_info
```

**Review by eteq — APPROVED** [review]
One *super* minor change to suggest.

I think this is fine regardless, but before merging I just want to double-check: @laurenmarietta did you try this new version locally and find it addressed your concerns? (it looks to me like yes, but I just want to double-check before merging)

**Comment by laurenmarietta** [comment]
Yes, the changes here fixed the error I was running into!

**Pull request merged**